server/ladders-local.ts

```diff
@@ -15,59 +15,58 @@
 
 'use strict';
 
-/** @type {typeof import('../lib/fs').FS} */
-const FS = require(/** @type {any} */('../.lib-dist/fs')).FS;
+import {FS} from '../lib/fs';
 
 // ladderCaches = {formatid: ladder OR Promise(ladder)}
 // Use Ladders(formatid).ladder to guarantee a Promise(ladder).
 // ladder is basically a 2D array representing the corresponding ladder.tsv
 //   with userid in front
-/** @typedef {[string, number, string, number, number, number, string]} LadderRow [userid, elo, username, w, l, t, lastUpdate] */
-/** @type {Map<string, LadderRow[] | Promise<LadderRow[]>>} formatid: ladder */
-let ladderCaches = new Map();
+/** [userid, elo, username, w, l, t, lastUpdate */
+type LadderRow = [string, number, string, number, number, number, string];
+/** formatid: ladder */
+type LadderCache = Map<string, LadderRow[] | Promise<LadderRow[]>>;
+
+const ladderCaches: LadderCache = new Map();
 
 class LadderStore {
-	/**
-	 * @param {string} formatid
-	 */
-	constructor(formatid) {
+	formatid: string;
+	ladder: LadderRow[] | null;
+	ladderPromise: Promise<LadderRow[]> | null;
+	saving: boolean;
+	static formatsListPrefix = '|,LL';
+	static ladderCaches = ladderCaches;
+
+	constructor(formatid: string) {
 		this.formatid = formatid;
-		/** @type {LadderRow[]?} */
 		this.ladder = null;
-		/** @type {Promise<LadderRow[]>?} */
 		this.ladderPromise = null;
+		this.saving = false;
 	}
 
-	/**
-	 * @return {Promise<LadderRow[]>}
-	 */
 	getLadder() {
 		if (!this.ladderPromise) this.ladderPromise = this.load();
 		return this.ladderPromise;
 	}
-	/**
-	 * Internal function, returns a Promise for a ladder
```

Comment:
Hey, why'd you delete this? The doc comment part is still important.

Reply:
Oh, that's my bad.